benchmarks/tests/test_partitioned_dataset_filter_benchmark.py

```diff
@@ -0,0 +1,54 @@
+import copy
+import json
+
+import pytest
+
+from .. import dataset_filter_benchmark
+from ..tests._asserts import assert_benchmark, assert_context, R_CLI
+
+
+HELP = """
+Usage: conbench dataset-filter [OPTIONS] SOURCE
+
+  Run partitioned-dataset-filter benchmark.
+
+Options:
+  --cpu-count INTEGER
+  --iterations INTEGER   [default: 1]
+  --gc-collect BOOLEAN   [default: true]
+  --gc-disable BOOLEAN   [default: true]
+  --show-result BOOLEAN  [default: true]
+  --show-output BOOLEAN  [default: false]
+  --run-id TEXT          Group executions together with a run id.
+  --help                 Show this message and exit.
+"""
+
+
+def assert_benchmark(result, source, name, case, language="Python"):
+    munged = copy.deepcopy(result)
+    expected = {
+        "name": name,
+        "dataset": source,
+        "cpu_count": None,
+        "gc_collect": True,
+        "gc_disable": True,
+    }
+    if language == "R":
+        del expected["gc_collect"]
+        del expected["gc_disable"]
+        expected["query"] = case[0]
+        expected["language"] = "R"
+    assert munged["tags"] == expected
+    assert_context(munged, language=language)
+
+
+benchmark = dataset_filter_benchmark.PartitionedDatasetFilterBenchmark()
+
+
+@pytest.mark.parametrize("case", benchmark.cases, ids=benchmark.case_ids)
+def test_partitioned_dataset_filter_one(case):
+    pytest.skip("needs a test partitioned dataset")
```

Comment:
Let's unskip this then, otherwise we won't know it's working.

For example, I just tried run this benchmark, and I got the following error, so perhaps the instructions and GitHub actions need updating. Plus coordination with Elena for benchmarks-jobs.

```
> library(arrowbench); run_one(arrowbench:::dataset_taxi_parquet, n_iter=1, cpu_count=NULL, query="small_no_files")
$error
[1] "Error in library(\"dplyr\") : there is no package called ‘dplyr’"
[2] "Calls: run_bm -> <Anonymous> -> library"                         
[3] "Execution halted"                                                
attr(,"status")
[1] 1

$params
$params$cpu_count
NULL

$params$query
[1] "small_no_files"

$params$name
[1] "dataset_taxi_parquet"


> 
>
[Previously saved workspace restored]

Error in library("dplyr") : there is no package called ‘dplyr’
Calls: run_bm -> <Anonymous> -> library
Execution halted
```

Reply:
And unskipping these will download around 40gb to travis/GHA whenever we run it without setting up caching of our data directory. We could setup that cache (if they allow that much data to be cached), or manufacture new benchmarks to test with that only use a portion of this data, but I'm not sure either of those are worth the time.

Reply:
> 🤔 that's odd — the instructions for arrowbench say to install with `remotes::install_github("ursacomputing/arrowbench", dependencies = TRUE)` but the ones in this repo don't have the `dependencies = TRUE` which will trigger dplyr to be installed (among other things). Want me to update the instructions in benchmarks in this PR?

Ah, I see. The arrowbench instructions didn't initially have `dependencies = TRUE`, and I must have written these instructions at that point in time. Either way, the GitHub action and the README need to be updated. Probably benchmark-jobs too.

That said, when I just tried with `dependencies = TRUE`, I got the following error. I'll update the README and the GitHub action in a different PR. Perhaps it will work there, and I'll debug my machine in a bit.

Error: package or namespace load failed for ‘fst’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/usr/local/Cellar/r/4.0.4/lib/R/library/00LOCK-fst/00new/fst/libs/fst.so':
  dlopen(/usr/local/Cellar/r/4.0.4/lib/R/library/00LOCK-fst/00new/fst/libs/fst.so, 6): Symbol not found: _XXH32
  Referenced from: /usr/local/Cellar/r/4.0.4/lib/R/library/00LOCK-fst/00new/fst/libs/fst.so
  Expected in: flat namespace
 in /usr/local/Cellar/r/4.0.4/lib/R/library/00LOCK-fst/00new/fst/libs/fst.so
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/usr/local/Cellar/r/4.0.4/lib/R/library/fst’
Error: Failed to install 'arrowbench' from GitHub:
  (converted from warning) installation of package ‘fst’ had non-zero exit status

Reply:
Yeah, IME fst can be finicky when installing it (there are some system dependencies, so if you're building from source those need to be troublshot, etc.). FST isn't actually necessary for any of the tests we'll be running on conbench, so maybe we should promote `dplyr` to a higher level of dependency and keep the install commands the way they are in benchmarks/conbench.

Reply:
That's what I've done, let's leave these installs as they are for now and not worry about trying to build/install other packages that we won't be testing on conbench. 

https://github.com/ursacomputing/arrowbench/commit/af2a2bf779ba158e31f10b2155e3a15422260f8c